main.ts

```diff
@@ -453,29 +459,29 @@ class Lexer {
       // table no leading pipe (gfm)
       if (top && (cap = this.rules.nptable.exec(src))) {
         item = {
-          type: 'table',
-          header: splitCells(cap[1].replace(/^ *| *\| *$/g, '')),
+          type: 'table' as 'table',
+          header: splitCells(cap[1].replace(/^ *| *\| *$/g, ''), 0),//NOTE: "0" added by me, check this if code does not work. Used to be nothing -> compile error //@olaven
           align: cap[2].replace(/^ *|\| *$/g, '').split(/ *\| */),
           cells: cap[3] ? cap[3].replace(/\n$/, '').split('\n') : []
-        };
+        } as Tokens.Table;
 
         if (item.header.length === item.align.length) {
           src = src.substring(cap[0].length);
 
           for (i = 0; i < item.align.length; i++) {
-            if (/^ *-+: *$/.test(item.align[i])) {
+            if (/^ *-+: *$/.test(item.align[i] as string)) {
               item.align[i] = 'right';
-            } else if (/^ *:-+: *$/.test(item.align[i])) {
+            } else if (/^ *:-+: *$/.test(item.align[i] as string)) {
               item.align[i] = 'center';
-            } else if (/^ *:-+ *$/.test(item.align[i])) {
+            } else if (/^ *:-+ *$/.test(item.align[i] as string)) {
               item.align[i] = 'left';
             } else {
               item.align[i] = null;
             }
           }
 
-          for (i = 0; i < item.cells.length; i++) {
-            item.cells[i] = splitCells(item.cells[i], item.header.length);
+          for (i = 0; i < item.cells.length; i++) { //NOTE: risky conversion with toString? -@olaven
+            item.cells[i] = splitCells(item.cells[i].toString(), item.header.length);
```

Comment:
Isn't `Tokens.Table` already dictates `cells` to be `string[][]`? Maybe explicitly mark `item` declaration to have type `Table`

Actually it seems that `Tokens.Table` type annotation of `cells` seems incorrect (it was copied from DefinitelyTyped) and should be `string[]`.

Reply:
This may be. However, `cells` is treated as `string[][]` other places in the code. E.g. it is expected to be `string[][]` in 484, as `splitCells` returns `string[]`. 

Perhaps there is something I am not catching here? 😃